### PR TITLE
Provides a mechanism to give the value of a key to the connect configuration. See issue #45

### PR DESCRIPTION
**Pull request by Leeacarroll**
## Summary
Adds the ability to provide a key value to connect configurations. This allows the use of a Kafka Config provider to provide key values for serverless use cases. Using a secure source for the config provider is essential. An example use-case is:

sink Connector -> Config provider -> AWS Secrets Manager Config Provider Implementation -> AWS secrets Manager Repo for storing and rotating keys

## Related Issue

#45 

## Relevant Documentation

See here for an example use-case: https://docs.aws.amazon.com/msk/latest/developerguide/mkc-debeziumsource-connector-example.html

## How Has This Been Tested?
Added integration test. ensures existing config does not break
UAT on staging MSK connect cluster

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Leeacarroll** [comment]
CLA Signed.

**Comment by Leeacarroll** [comment]
Some notes on security: I've noticed the logging logs all parameters regardless of sensitivity, is that correct?

File: streaming-connect-sink/src/test/resources/aep-connector-with-jwt-token.json [hunk 1]
```diff
@@ -14,5 +14,5 @@
   "aep.connection.auth.client.secret": "client_secret",
   "aep.connection.auth.imsOrg": "imsOrg@AdobeOrg",
   "aep.connection.auth.accountKey": "technical-account-id",
-  "aep.connection.auth.filePath": "%s"
+  "aep.connection.auth.filePath": "C:/Users/u207754/Documents/git/ba-experience-platform-streaming-connect/BA-experience-platform-streaming-connect/streaming-connect-sink/build/resources/test/secret.key"
```
[the remark this answers is not in the document]
Comment (Leeacarroll) [reply streaming-connect-sink/src/test/resources/aep-connector-with-jwt-token.json:17]: my bad, on windows and was lazy so just removed the slash at the start. thought i'd reverted back to %s. probably should have made it os agnostic!

**Comment by Leeacarroll** [comment]
Hi Just working through system testing and getting the below error. Basically my local produced zip distrubution file for the connector fails when used to deploy across a msk cluster due to not being able to find the connector impl class (see error below). I'm building the zip archive using clean build kafkaConnectPublish. This does not produce a single directory (which i think a plugin needs and the distro on clonfluent for the plugin has. However even when i mannually add this it still fails

error


[Worker-048b5bf1cde2122fd] org.apache.kafka.connect.errors.ConnectException: Failed to find any class that implements Connector and which name matches com.adobe.platform.streaming.sink.impl.AEPSinkConnector, available  connectors are: PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSinkConnector, name='org.apache.kafka.connect.file.FileStreamSinkConnector', version='2.7.1', encodedVersion=2.7.1, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class  org.apache.kafka.connect.file.FileStreamSourceConnector, name='org.apache.kafka.connect.file.FileStreamSourceConnector', version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorCheckpointConnector,  name='org.apache.kafka.connect.mirror.MirrorCheckpointConnector', version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorHeartbeatConnector, name='org.apache.kafka.connect.mirror.MirrorHeartbeatConnector',  version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorSourceConnector, name='org.apache.kafka.connect.mirror.MirrorSourceConnector', version='1', encodedVersion=1,  type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockConnector, name='org.apache.kafka.connect.tools.MockConnector', version='2.7.1', encodedVersion=2.7.1, type=connector, typeName='connector', location='classpath'},  PluginDesc{klass=class org.apache.kafka.connect.tools.MockSinkConnector, name='org.apache.kafka.connect.tools.MockSinkConnector', version='2.7.1', encodedVersion=2.7.1, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSourceConnector,  name='org.apache.kafka.connect.tools.MockSourceConnector', version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.SchemaSourceConnector, name='org.apache.kafka.connect.tools.SchemaSourceConnector',  version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSinkConnector, name='org.apache.kafka.connect.tools.VerifiableSinkConnector', version='2.7.1', encodedVersion=2.7.1,  type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSourceConnector, name='org.apache.kafka.connect.tools.VerifiableSourceConnector', version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source',  location='classpath'} | [Worker-048b5bf1cde2122fd] org.apache.kafka.connect.errors.ConnectException: Failed to find any class that implements Connector and which name matches com.adobe.platform.streaming.sink.impl.AEPSinkConnector, available connectors are: PluginDesc{klass=class  org.apache.kafka.connect.file.FileStreamSinkConnector, name='org.apache.kafka.connect.file.FileStreamSinkConnector', version='2.7.1', encodedVersion=2.7.1, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSourceConnector,  name='org.apache.kafka.connect.file.FileStreamSourceConnector', version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorCheckpointConnector, name='org.apache.kafka.connect.mirror.MirrorCheckpointConnector',  version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorHeartbeatConnector, name='org.apache.kafka.connect.mirror.MirrorHeartbeatConnector', version='1', encodedVersion=1,  type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorSourceConnector, name='org.apache.kafka.connect.mirror.MirrorSourceConnector', version='1', encodedVersion=1, type=source, typeName='source',  location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockConnector, name='org.apache.kafka.connect.tools.MockConnector', version='2.7.1', encodedVersion=2.7.1, type=connector, typeName='connector', location='classpath'}, PluginDesc{klass=class  org.apache.kafka.connect.tools.MockSinkConnector, name='org.apache.kafka.connect.tools.MockSinkConnector', version='2.7.1', encodedVersion=2.7.1, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSourceConnector,  name='org.apache.kafka.connect.tools.MockSourceConnector', version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.SchemaSourceConnector, name='org.apache.kafka.connect.tools.SchemaSourceConnector',  version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSinkConnector, name='org.apache.kafka.connect.tools.VerifiableSinkConnector', version='2.7.1', encodedVersion=2.7.1,  type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSourceConnector, name='org.apache.kafka.connect.tools.VerifiableSourceConnector', version='2.7.1', encodedVersion=2.7.1, type=source, typeName='source',  location='classpath'}
-- | --

File: streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/AbstractSinkConnector.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
 package com.adobe.platform.streaming.sink;
 
 import com.google.common.collect.ImmutableMap;
+import kafka.server.ConfigType;
```
[the remark this answers is not in the document]
Comment (Leeacarroll) [reply streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/AbstractSinkConnector.java]: removed

File: streaming-connect-sink/src/test/resources/aep-connector-with-jwt-token.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Leeacarroll) [reply streaming-connect-sink/src/test/resources/aep-connector-with-jwt-token.json]: replaced

File: streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/AbstractSinkConnector.java [hunk 3]
```diff
@@ -60,6 +61,11 @@ public void stop() {
 
   @Override
   public ConfigDef config() {
+
```
Comment (vivetiwa) [inline streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/AbstractSinkConnector.java:64]: Redundant code please remove.
```
ConfigDef configDef = new ConfigDef();
    configDef.define("aep.connection.auth.keyValue", ConfigDef.Type.PASSWORD,null,null,"")
```

**Comment by Leeacarroll** [comment]
Hi, Just seen this failed on some code hygiene. Did the logging issue rasied get sorted. I'm still a little nervous about the logging. I've took a different approach locally to get round this and have implement the ability to specify a s3 location for keys. this removes the possibility of config logging leaking data. I can submit that code as well.

**Comment by Leeacarroll** [comment]
Given we now can use: token-type = "oauth2_access_token" without the need for key stores this PR is redundant and can be closed.

**Pull request closed without merging**